Sources/CacheAdvance/EncodableInteger.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by Dan Federman on 12/3/19.
+//  Copyright © 2019 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A protocol representing an integer that can be encoded as data.
+protocol EncodableInteger {
```

Comment:
I'm not sure I follow this. The name [encodable](https://developer.apple.com/documentation/swift/encodable) [integer](https://developer.apple.com/documentation/swift/int) makes it feel like you're "making an integer encodable" (which we know this is always true).

Then in the description you talk about being able to represent an Int as Data, but there's no reference to any Data in the protocol signature.

From what I can tell this protocol is mostly to convert to manipulate the big-endian of an instance?

Reply:
This protocol defines a type that:
- Can be encoded as `Data` of width `storageLength`
- Can be initialized back from `Data` of width `storageLength`
- Can be encoded in `bigEndian` format and read back out in the host's endian preference.

To be fair, any `integer` can do this, but this protocol allows me to add conveniences. The good news is that it is an `internal` protocol (for now). I'm open to different naming given the above.

Also, I 100% agree this should be `ExpressibleByIntegerLiteral`. I'd forgotten that existed.

Reply:
Okay I think I can simplify the naming by using `FixedWidthInteger`. Commit incoming.

Reply:
https://github.com/dfed/CacheAdvance/pull/15/commits/f6475da880df78eb74fdecd09a52b990d7519bc6 should make the naming a bit more clear